documentation/add/11a-sqw-compression-implement.md

```diff
@@ -0,0 +1,116 @@
+present.
+
+These methods will be called at the point of reading or writing.
+
+## Locations for compression and decompression
+
+The locations here are those for the *current* faccess default faccess_sqw_v4. It is
+intended that they will port to equivalent locations in the new faccess_sqw_v5 unless
+specifically noted.
+
+### Compression
+
+These locations are the locations where the pixel data is written; it appears that there
+is only one such location.
+
+`put_pix` is the method in horace_core\sqw\file_io\@faccess_sqw_v4\put_pix.m
+It uses 2 calls to fwrite: 
+   - at line 186 pixels *not* stored in a subclass object of PixelDataBase are written
+     by a single fwrite. Compression should be done just before this line. As this will require
+     additional memory of order a third of the original pixel data size, it may be
+     desirable to page this into smaller chunks to prevent problems with allocation of
+     memory.
+   - at line 176 pixels which *are* stored in a subclass object of PixelDataBase
+     are written by an fwrite per page. Again, compression should be done just before
+     this line. For the moment it is considered that the page size is sufficient to
+     control memory usage.
+
+### Decompression
+
+These locations are the locations where the file is read. There are more that one such 
+locations.
+
+`get_pix_` is the method in horace_core\sqw\file_io\@faccess_sqw_v4\put_pix.m. It is
+called from get_raw_pix. Note that another method `get_pix` exists as an alternative
+caller of `get_pix_` but it appears only to be used for testing. This performs is single
+fread and appears to correspond to the compression case where the data written is not
+from a PixelDataBase subclass. The decompression should be done immediately after the
+fread. Any additional paging done on compression should also be done here.
+
+Note that this is also the method used for PixelDataMemory objects, where the method
+`init_from_file_accessor_` calls `get_raw_pix` and hence `get_pix`. It is unclear how
+the paging done in `put_pix` above for PixelDataMemory objects (included in the general
+processing for PixelDataBase objects) is mirrored here where there is only a single fread.
+
+For PixelDataFileBacked objects, the method `init_from_file_accessor_` is specialised
+to used memmapfile. This is apparently an import from python which is less succesful 
+in matlab. Critically here the decompression process breaks the symmetry of data size
+between memory and file: the pixel record size here is now still 9 in memory but 3 on
+disk. Consequently the memmapfile method cannot now obviously be used, and the older
+pre-v4 straight read used instead. The symmetry between formats on reading and writing
+should still be honoured.
+
+### Other locations
+
+Note that the get and put methods noted above in faccess_sqw_v4 are duplicated in
+sqw_file_formatters in the directories below. It is not clear if they are used; they do
```

Comment:
They are not duplicated -- `sqw_file_formatters` describes common interface to pixel access methods and `faccess_vX` implements actual operations